### PR TITLE
issue/1184 – Establish core objects for affiliates, referrals, creatives, and visits

### PR DESCRIPTION
**Pull request by DrewAPicture**
Issue: #1184 

This PR only handles introducing the new object classes. For implementation, see #1301 (PR: #1302).

### CODE REVIEW

File: includes/objects/class-affwp-affiliate.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
+<?php
+
+final class AffWP_Affiliate extends AffWP_Object {
```
Comment (pippinsplugins) [inline includes/objects/class-affwp-affiliate.php]: Let's plan to keep `final` for a version or two and then remove it once we're sure it's safe to extend.

File: includes/objects/class-affwp-affiliate.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	 * Object type.
+	 *
+	 * Used as the cache group and for accessing object DB classes in the parent.
+	 *
+	 * @since 1.9
+	 * @access public
+	 * @static
+	 * @var string
+	 */
+	public static $object_type = 'affiliates';
```
Comment (pippinsplugins) [inline includes/objects/class-affwp-affiliate.php]: Can we make this singular?

Comment (pippinsplugins) [inline includes/objects/class-affwp-affiliate.php]: Same with the other object ypes.

File: includes/objects/class-affwp-affiliate.php (hunk 1 above)
Comment (DrewAPicture) [inline includes/objects/class-affwp-affiliate.php]: Roger.

**Pull request merged**